agent/agent.go

```diff
@@ -269,6 +269,17 @@ func (a *Agent) Start() error {
 		return fmt.Errorf("Failed to setup node ID: %v", err)
 	}
 
+	// Warn if the node name is incompatible with DNS
+	if InvalidDnsRe.MatchString(a.config.NodeName) {
+		a.logger.Printf("[WARN] agent: Node name %q will not be discoverable "+
+			"via DNS due to invalid characters. Valid characters include "+
+			"all alpha-numerics and dashes.", a.config.NodeName)
+	} else if ( len(a.config.NodeName) > MaxDNSLabelLength ) {
```

Comment:
Could you run `make format` in your branch - it runs `go fmt` which should reduce this (and a couple other places) to something like:  
```
if len(a.config.NodeName) > MaxDNSLabelLength {
```